torch/ao/quantization/experimental/quantizer.py

```diff
@@ -1,12 +1,58 @@
+import torch
 from torch import Tensor
+from torch.ao.quantization.experimental.observer import APoTObserver
+from torch.ao.quantization.experimental.apot_utils import float_to_apot
 
 # class to store APoT quantizer
 # implements quantize and dequantize
 # and stores all quantization parameters
 class APoTQuantizer():
-    @staticmethod
-    def quantize_APoT(tensor2quantize: Tensor) -> Tensor:
-        raise NotImplementedError
+    b: int
+    k: int
+    n: int
+    signed: bool
+    quantization_levels: torch.Tensor
+    level_indices: torch.Tensor
+    data: torch.Tensor
+
+    def __init__(
+        self,
+        b,
+        k,
+        max_val,
+        signed,
+            dtype=torch.quint8) -> None:
+        self.signed = signed
+
+        # check for valid inputs of b, k
+        assert(k and k != 0)
+        assert(b % k == 0)
+        self.b = b
+        self.k = k
+        self.n = b // k
+
+        # make observer, get quantizion levels and level indices
+        obs = APoTObserver(max_val=max_val, b=b, k=k)
+        obs_result = obs.calculate_qparams(signed=signed)
```

Comment:
I feel this observer is not used the same way as the other observers in the code base, since it does not observe anything, is `max_val` something we get from observed Tensor? would it make more sense to move this argument to `calculate_qparams`? meaning that we are overriding the `max_val` instead from the internal state of observer while calculating quantization paramters.

Reply:
yes good point, I can make a new PR to modify so calculate_qparams is the one that takes in max_val